File: .android/Flutter/src/main/kotlin/com/salesmate_chat/UserDetails.kt
package com.salesmate_chat

class UserDetails(
    var email: String,
    var firstName: String,
    var lastName: String,
    var userHash: String,
) {

    fun toMap(): Map<String, String> = mapOf(
        "email" to email, "firstName" to firstName, "lastName" to lastName, "userHash" to userHash
    )
}

File: .ios/Runner/UserDetails.swift
import Foundation

@objc public class UserDetails: NSObject {
    @objc public var email: String
    @objc public var firstName: String
    @objc public var lastName: String
    @objc public var userHash: String
    
    @objc public init(email: String, firstName: String, lastName: String, userHash: String) {
        self.email = email
        self.firstName = firstName
        self.lastName = lastName
        self.userHash = userHash
        super.init()
    }
    
    public func toMap() -> [String: String] {
        return [
            "email": email,
            "firstName": firstName,
            "lastName": lastName,
            "userHash": userHash
        ]
    }
}
